src/Microsoft.Azure.EventHubs.Processor/PartitionManager.cs

```diff
@@ -299,7 +305,12 @@ internal PartitionManager(EventProcessorHost host)
 
                 try
                 {
-                    await Task.Delay(leaseManager.LeaseRenewInterval, cancellationToken).ConfigureAwait(false);
+                    // Consider reducing the wait time with last lease-walkthrough's time taken.
+                    var elapsedTime = loopStopwatch.Elapsed;
+                    if (leaseManager.LeaseRenewInterval > elapsedTime)
+                    {
+                        await Task.Delay(leaseManager.LeaseRenewInterval.Subtract(elapsedTime), cancellationToken).ConfigureAwait(false);
```

Comment:
What will happen if subtract returns negative value? Delay will throw an exception I think. Can you please check it and make sure code address it?

Reply:
I considered that, below is the check to avoid that actually.
if (leaseManager.LeaseRenewInterval > elapsedTime)

Reply:
You would need to calculate a different first and use it in the Delay call instead of comparing & subtract.